src/main/java/com/teragrep/jos_01/procfs/status/Statm.java

```diff
@@ -0,0 +1,109 @@
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ *
+ * Additional permission under GNU Affero General Public License version 3
+ * section 7
+ *
+ * If you modify this Program, or any covered work, by linking or combining it
+ * with other code, such other code is not for that reason alone subject to any
+ * of the requirements of the GNU Affero GPL version 3 as long as this Program
+ * is the same Program as licensed from Suomen Kanuuna Oy without any additional
+ * modifications.
+ *
+ * Supplemented terms under GNU Affero General Public License version 3
+ * section 7
+ *
+ * Origin of the software must be attributed to Suomen Kanuuna Oy. Any modified
+ * versions must be marked as "Modified version of" The Program.
+ *
+ * Names of the licensors and authors may not be used for publicity purposes.
+ *
+ * No rights are granted for use of trade names, trademarks, or service marks
+ * which are in The Program if any.
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+package com.teragrep.jos_01.procfs.status;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+// Provides information about memory usage, measured in pages.
+// Some of these values are inaccurate because of a kernel-internal scalability optimization.
+// If accurate values are required, use smaps or smaps_rollup instead, which are much slower but provide accurate, detailed information
+public class Statm implements Status {
+
+    private final ArrayList<String> rows;
+    private final LocalDateTime timestamp;
+    private final Map<String, String> statistics;
+
+    private enum fields {
+        size, resident, shared, text, lib, data, dt;
+    };
+
+    public Statm(ArrayList<String> rows) {
+        this.rows = rows;
+        statistics = new LinkedHashMap<String, String>();
+        Pattern pattern = Pattern
+                .compile(
+                        "(?<size>\\d+) (?<resident>\\d+) (?<shared>\\d+) (?<text>\\d+) (?<lib>\\d+) (?<data>\\d+) (?<dt>\\d+)"
+                );
+        for (String row : rows) {
+            Matcher matcher = pattern.matcher(row);
+            if (matcher.find()) {
+                for (int i = 0; i < matcher.groupCount(); i++) {
+                    statistics
+                            .put(fields.values()[i].name(), matcher.group(fields.values()[i].name().replace("_", "")));
+                }
+            }
```

Comment:
This *MIGHT* be the same as the other `while(matcher.find())` comment but I am not sure.

Reply:
This one has the same formatting with multiple values in a single row